### PR TITLE
fix(pagerduty): Add Pocket PagerDuty module

### PR DESCRIPTION
**Pull request by kkyeboah**
## Goal

Create a Pocket Pagerduty construct

### CODE REVIEW

**Comment by jpetto** [comment]
i like this general pattern of having the constructor call a bunch of smaller functions to build up the necessary resources.

my question here, and in general in this repo, is should the smaller functions (e.g. `createSnsTopic`) be testable individually? should our focus be on testing these smaller functions instead of testing the class as a whole (which creates multiple resources)? i'm leaning towards the former - testing only the smaller functions, as if they behave consistently, we shouldn't have to test the full class/constructor. the problem here is, of course, the necessity of `this` when creating every resource.

**Comment by kkyeboah** [comment]
The context of what we are doing here should be taken into consideration. Individually testable functions are great but in this context, testing these functions provide little value. I say that because the individual functions could be tested to show they work perfectly but without testing the `custom construct`/class we are creating, we can't know if we will get the desired outcome when we use it. 

In the end, what I believe we are most interested in with this repository is the class being able to create all the resources, and in the appropriate order, to be useful to us as a single module: We can only verify this by testing the class itself.

**Comment by jpetto** [comment]
you bring up a good point about context, and i acknowledge i may be falling into a premature optimization trap here. (e.g. technically creating an SNS topic subscription doesn't need to know anything about PagerDuty, but practically this is how we are using it right now. we can refactor this later if we have to.)

the issue with testing any of this is that the only _real_ way to know if we're getting the desired outcome is to run `terraform apply`. otherwise we can only check for valid terraform - which is useful as a start, but doesn't actually tell us if the terraform we've generated will 1) actually work, and 2) achieve the expected result in terms of functioning infrastructure. i don't know a way around this without a deploy to a testing environment.

that all to say, i'm good with this approach you have here. i think the smaller functions here are great and should be repeated in the other modules.

File: src/pocket/PocketPagerDuty.ts [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+      cloudwatchNonCriticalIntegration,
+      PAGERDUTY_SERVICE_URGENCY.NON_CRITICAL
+    );
+  }
+
+  private createSnsTopicSubscription(
+    topic: SnsTopic,
+    integration: ServiceIntegration,
+    urgency: PAGERDUTY_SERVICE_URGENCY
+  ) {
```
Comment (jpetto) [inline src/pocket/PocketPagerDuty.ts:107]: can we have return types on these functions?

**Review by jpetto — CHANGES_REQUESTED** [review]
let's add return types to the functions? otherwise lgtm.

File: src/pocket/PocketPagerDuty.ts (hunk 1 above)
Comment (kkyeboah) [reply src/pocket/PocketPagerDuty.ts]: done

**Review by jpetto — APPROVED** [review]

**Pull request merged**